### ISSUE TITLE
Make escaping less context sensitive and error prone

### ISSUE THREAD

**Issue by leonard84** [asciidoctor/asciidoctor#3567]
Currently the escape mechanism is context sensitive you can't just always write `\*` and get `*`, it depends on whether there is another lone `*` somewhere in the vicinity and you can't even escape both `*` the same way. I'd suggest to treat `\*` always as a literal `*`, and for the very few times that you want to write `\*` just write `\\*`.

```
To use Spock in your Groovy 3 project just select the `spock-\*-2.0-M2` artifact(s) ending with `-groovy-3.0`.
```
renders as 
>  To use Spock in your Groovy 3 project just select the spock-\\*-2.0-M2 artifact(s) ending with -groovy-3.0.

Whereas 
```It is *very important* to *do not mix* the `spock-\*-2.x-groovy-2.5` artifacts with the `groovy-*-3.x` artifacts on a classpath.```
Renders as
>It is very important to do not mix the spock-\*-2.x-groovy-2.5 artifacts with the groovy-*-3.x artifacts on a classpath.

**Comment by mojavelinux** [comment]
I agree with what you are saying. This is a duplicate of #901 (and several of the linked issues). It cannot easily be solved with the current regex-based parser. So to do it right, it's blocked by #61.
